procurement_for_service/models/sale_order.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+from openerp import models, api, exceptions, _
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.one
+    def action_button_confirm(self):
+        procurement_obj = self.env['procurement.order']
+        res = super(SaleOrder, self).action_button_confirm()
+        for line in self.order_line:
+            routes = line.product_id.route_ids.filtered(
+                lambda r: r.name in ('Make To Order', 'Buy'))
+            if line.product_id.type == 'service' and len(routes) == 2:
```

Comment:
Aparte.. nos estamos liando. Sale_service ahora, en esta versión V8, solo
genera tasks si el campo generate task del producto está marcado. No va por
abastecimientos.
¿Deberiamos incluir nosotros esa opción? Que si es servicio, manufacture
genere task? O no?

2015-08-03 12:28 GMT+02:00 Ana Juaristi <ajuaristio@gmail.com>:

> Eso es... como el sale_service ya lo hace para los que están marcados como
> manufacture, el nuestro lo haría solo para buy.
> De hecho, quizás esto debería ir dentro del sale_service pero la cosa es
> que manufacture tiene dependencia con mrp y en este caso no sería
> necesario.
> La cosa es que cuando es buy-make to order debe generar una compra de
> servicio. El módulo cubre esta necesidad sin tener en cuenta nada más.
>
>
> 2015-08-03 12:25 GMT+02:00 Pedro M. Baeza <notifications@github.com>:
>
>> In procurement_for_service/models/sale_order.py
>> <https://github.com/odoomrp/odoomrp-wip/pull/906#discussion_r36072823>:
>>
>> > +# For copyright and license notices, see __openerp__.py file in root directory
>> > +##############################################################################
>> > +from openerp import models, api, exceptions, _
>> > +
>> > +
>> > +class SaleOrder(models.Model):
>> > +    _inherit = 'sale.order'
>> > +
>> > +    @api.one
>> > +    def action_button_confirm(self):
>> > +        procurement_obj = self.env['procurement.order']
>> > +        res = super(SaleOrder, self).action_button_confirm()
>> > +        for line in self.order_line:
>> > +            routes = line.product_id.route_ids.filtered(
>> > +                lambda r: r.name in ('Make To Order', 'Buy'))
>> > +            if line.product_id.type == 'service' and len(routes) == 2:
>>
>> Cómo lo hace el actual módulo sale_service?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/odoomrp/odoomrp-wip/pull/906/files#r36072823>.
>>
>
>
>
> --
> CEO Avanzosc, S.L <http://www.avanzosc.es> : Office phone / Tfono
> oficina: (+34) 943 02 69 02
> Ana Juaristi Olalde <http://www.anajuaristi.com/>: Personal phone: 677 93
> 42 59. User/usuario skype: Avanzosc
> www.openerpsite.com
>
>
> *El contenido de esta comunicación y de toda su documentación anexa es
> confidencial y se dirige exclusivamente a su destinatario. El uso no
> autorizado de esta información está prohibido por la legislación vigente.
> Si usted no es el destinatario le rogamos nos lo indique, no comunique su
> contenido a terceros y proceda a su destrucción. Disculpe las molestias que
> le haya ocasionado la recepción indebida de este e-mail. Sus datos figuran
> en un fichero cuyo titular es Avanzosc, S.L., a quien usted puede dirigirse
> para ejercer sus derechos de acceso, rectificación, cancelación y oposición
> en Klara Donea 13, 20720, Azkoitia (Gipuzkoa), Tef. 943 02 69 02 - **administracion@avanzosc.
> <soporte@avanzosc.es>es*
>
> *Komunikazio honen edukia eta dokumentazio erantsia konfidentziala da eta
> hartzaileak bakarrik jaso beharko luke. Indarrean dagoen legeriak debekatu
> egiten du bertan eskainitako informazioa baimenik gabe erabiltzea.
> Komunikazioa zuri iritsi bazaizu, baina zu ez bazara hartzailea, mesedez,
> guri jakinarazi, eta jasotako informazioa ez inori jakinarazi eta suntsitu.
> Barkatu okerreko email hau jasotzeak eragindako eragozpenak. Zure datuak
> Avanzosc, S.L. enpresaren fitxategietan sartuta daude. Zure datuak atzitzea
> eska dezakezu, bai eta, datuak zuzentzea, ezereztea eta tratamenduari aurka
> egitea ere. Horretarako, enpresara jo dezakezu, helbide honetan: **Klara
> Donea 13** 20720, Azkoitia (Gipuzkoa), telefonoa: 943 02 69 02 - *
> *administracion@avanzosc. <soporte@avanzosc.es>es* *This message and all
> documents attached to it are confidential and intended only for the person
> or entity to which it is addressed. Any use of this information by
> unauthorised persons is prohibited under current legislation. If you
> received this message by error, please advise us, destroy it and refrain
> from communicating its contents to third parties. We apologise for any
> inconvenience receiving this email improperly may cause to you. Your
> personal data are included in a file owned by Avanzosc, S.L. If you want to
> exercise your rights of access, correction, erasure and objection you can
> contact the Controller at **Klara Donea 13** 20720, Azkoitia (Gipuzkoa),
> T: 943 02 69 02 – administracion@avanzosc. <soporte@avanzosc.es>es*
>



-- 
CEO Avanzosc, S.L <http://www.avanzosc.es> : Office phone / Tfono oficina:
(+34) 943 02 69 02
Ana Juaristi Olalde <http://www.anajuaristi.com/>: Personal phone: 677 93
42 59. User/usuario skype: Avanzosc
www.openerpsite.com


*El contenido de esta comunicación y de toda su documentación anexa es
confidencial y se dirige exclusivamente a su destinatario. El uso no
autorizado de esta información está prohibido por la legislación vigente.
Si usted no es el destinatario le rogamos nos lo indique, no comunique su
contenido a terceros y proceda a su destrucción. Disculpe las molestias que
le haya ocasionado la recepción indebida de este e-mail. Sus datos figuran
en un fichero cuyo titular es Avanzosc, S.L., a quien usted puede dirigirse
para ejercer sus derechos de acceso, rectificación, cancelación y oposición
en Klara Donea 13, 20720, Azkoitia (Gipuzkoa), Tef. 943 02 69 02 -
**administracion@avanzosc.
<soporte@avanzosc.es>es*

*Komunikazio honen edukia eta dokumentazio erantsia konfidentziala da eta
hartzaileak bakarrik jaso beharko luke. Indarrean dagoen legeriak debekatu
egiten du bertan eskainitako informazioa baimenik gabe erabiltzea.
Komunikazioa zuri iritsi bazaizu, baina zu ez bazara hartzailea, mesedez,
guri jakinarazi, eta jasotako informazioa ez inori jakinarazi eta suntsitu.
Barkatu okerreko email hau jasotzeak eragindako eragozpenak. Zure datuak
Avanzosc, S.L. enpresaren fitxategietan sartuta daude. Zure datuak atzitzea
eska dezakezu, bai eta, datuak zuzentzea, ezereztea eta tratamenduari aurka
egitea ere. Horretarako, enpresara jo dezakezu, helbide honetan: **Klara
Donea 13** 20720, Azkoitia (Gipuzkoa), telefonoa: 943 02 69 02 - *
*administracion@avanzosc. <soporte@avanzosc.es>es* *This message and all
documents attached to it are confidential and intended only for the person
or entity to which it is addressed. Any use of this information by
unauthorised persons is prohibited under current legislation. If you
received this message by error, please advise us, destroy it and refrain
from communicating its contents to third parties. We apologise for any
inconvenience receiving this email improperly may cause to you. Your
personal data are included in a file owned by Avanzosc, S.L. If you want to
exercise your rights of access, correction, erasure and objection you can
contact the Controller at **Klara Donea 13** 20720, Azkoitia (Gipuzkoa), T:
943 02 69 02 – administracion@avanzosc. <soporte@avanzosc.es>es*